### PR TITLE
Tentatively comment out GCC TSAN in CI to avoid blocking PRs

### PR DESCRIPTION
**Pull request by git-hulk**
Currently, GCC TSAN was broken but didn't identify the underlying reason,
so let's comment it out first to avoid blocking new PRs and add it back
after it was fixed.

#1662 is tracking the GCC TSAN issue.

### CODE REVIEW

**Comment by git-hulk** [comment]
#1663 has removed this unit, close this PR.

**Comment by enjoy-binbin** [comment]
The topic of #1663 is not related to this, i think it can be kept

**Review by enjoy-binbin — DISMISSED** [review]

**Comment by tisonkun** [comment]
We should comment out and add an inline comment. Even we should open an issue to dig out, instead of remove the matrix item.

**Comment by tisonkun** [comment]
This is the related issue - https://github.com/apache/kvrocks/issues/1662

**Comment by enjoy-binbin** [comment]
why bring this into that typo ci? we can merge this one first and then refresh the typo ci one

**Comment by tisonkun** [comment]
@enjoy-binbin you're right.

File: .github/workflows/kvrocks.yaml [hunk 1]
```diff
@@ -134,12 +134,6 @@ jobs:
             with_sanitizer: -DENABLE_ASAN=ON
             without_jemalloc: -DDISABLE_JEMALLOC=ON
             compiler: clang
```
Comment (tisonkun) [inline .github/workflows/kvrocks.yaml:136]:
```suggestion
            compiler: clang
          # FIXME - tsan.so not found https://github.com/apache/kvrocks/issues/1662
          # - name: Ubuntu GCC TSan
          #   os: ubuntu-20.04
          #   without_jemalloc: -DDISABLE_JEMALLOC=ON
          #   with_sanitizer: -DENABLE_TSAN=ON
          #   compiler: gcc
          #   ignore_when_tsan: -tags="ignore_when_tsan"
```

**Review by tisonkun — APPROVED** [review]

**Review by enjoy-binbin — APPROVED** [review]

**Pull request merged**